students/LouReis/Lesson05/mailroom3.py

```diff
@@ -0,0 +1,173 @@
+# mailroom2.py
+# Coded by LouReis
+
+"""
+Write a small command-line script called mailroom.py.
+It should have a data structure that holds a list of your donors and a history
+of the amounts they have donated. This structure should be populated at first
+with at least five donors, with between 1 and 3 donations each.
+The script should prompt the user (you) to choose from a menu of 3 actions:
+“Send a Thank You”, “Create a Report” or “quit”)
+
+Report should look like the following:
+Donor Name                | Total Given | Num Gifts | Average Gift
+------------------------------------------------------------------
+Joe Donor                  $  653784.49           2  $   326892.24
+John Rich                  $   16396.10           3  $     5465.37
+Sally Neighbor             $     877.33           1  $      877.33
+Mack Jack                  $     708.42           3  $      236.14
+
+"""
+
+# donations = ['Robin Hood', 1500000, 3, 500000, 'Tycoon Reis', 75000000, 3, 25000000, 'Howie Long', 100000, 1, 100000, 'Joe Neighbor', 50, 2, 25, 'Rick Retiree', 1.00, 2, 0.50]
+# Data structure in global namespace to store all donations & donors.
+# donations = ['Robin Hood', 50000, 'Tycoon Reis', 25000000, 'Howie Long', 100000, 'Joe Neighbor', 25, 'Rick Retiree', 0.50, 'Robin Hood', 50000, 'Tycoon Reis', 25000000, 'Joe Neighbor', 25, 'Rick Retiree', 0.50, 'Robin Hood', 50000, 'Tycoon Reis', 25000000]
+donations = {"Robin Hood": [50000, 50000, 50000], "Tycoon Reis": [25000000, 25000000, 25000000], "Howie Long": [100000], "Joe Neighbor": [25, 25], "Rick Retiree": [0.50, 0.50]}
+
+# Tested & modified the below code that works for printed an unsorted report.
+# def calculate_sort(donations):
+#     for x, name in enumerate(donations):
+#         amounts = []
+#         total = 0
+#         count = 0
+#         amounts = donations.get(name)
+#         count = len(amounts)
+#         for item in amounts:
+#             total = total + item
+#         print ('{:25} ${:>15,.2f} {:>15} ${:>15,.2f}'.format(name, total, count, total/count))
+
+# Tested & modified the below code to develop the complex sorted output.
+# for key,value in sorted(donations.items(),key=lambda i:sum(i[1]),reverse=True):
+#     print (key,value)
+#     print (donations[key])
+#     total = 0
+#     total = sum(donations[key])
+#     print(total)
+
+# Below is the main menu function that continues prompting until quit.
+def main_menu(main_prompt,menu_options_dict):
+    while True:
+        try:
+            response = input(main_prompt)
+            menu_options_dict[response]()
+        except KeyError:
+            print("\n\n----------------PLEASE TRY AGAIN! PLEASE ENTER A VALID VALUE!----------------\n\n")
+            print("\n\n----------------PLEASE TRY AGAIN! PLEASE ENTER A VALID VALUE!----------------\n\n")
+            print("\n\n----------------PLEASE TRY AGAIN! PLEASE ENTER A VALID VALUE!----------------\n\n")
+            main_menu(main_prompt,menu_options_dict)
```

Comment:
no need for this line, your loop will continue looping - here you're creating another nested while True loop